src/config.rs

```diff
@@ -166,6 +167,26 @@ impl IndexMut<usize> for ColorList {
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct Shell<'a> {
+    program: Cow<'a, str>,
+    args: Vec<String>,
+}
+
+impl<'a> Shell<'a> {
+    pub fn new(program: &'a str) -> Shell<'a> {
```

Comment:
The following is informational since you mentioned not having worked with `Cow` before. Don't worry about making changes (the flexibility won't be needed). `Cow` is just really cool and I love telling people about it.

Consider the following signature:

```rust
pub fn new<'b, P>(program: P) -> Shell<'b>
    where P: Into<Cow<'b, str>>
```

With this, a `Shell` could be constructed with either an owned or borrowed string. For example,

```rust
let shell = Shell::new(String::new("/bin/bash")); // owned String
let shell = Shell::new("/bin/bash");  // &'static str
let shell_path = String::new("/bin/bash");
let shell = Shell::new(&shell_path[..]); // &str -- cannot outlive shell_path
```

Would all compile just fine.